sdk/lib/core/map.dart

```diff
@@ -261,15 +318,35 @@ abstract class Map<K, V> {
   /// and adds the key with the returned value to the map.
   ///
   /// If the key is not present, [ifAbsent] must be provided.
+  /// ```dart
+  /// final data = {1: 'A', 2: 'B', 3: 'C'};
+  /// data.update(2, (value) => 'ABCD', ifAbsent: () => 'G');
+  /// print(data); // {1: A, 2: ABCD, 3: C}
+  /// // if there is no key then ifAbsent is used
+  /// data.update(10, (value) => 'XYZ', ifAbsent: () => 'X');
+  /// print(data); // {1: A, 2: ABCD, 3: C, 10: X}
+  /// ```
   V update(K key, V update(V value), {V ifAbsent()?});
 
   /// Updates all values.
   ///
   /// Iterates over all entries in the map and updates them with the result
   /// of invoking [update].
+  /// ```dart
+  /// final data = {1: 'A', 2: 'B', 3: 'C'};
+  /// data.updateAll((key, value) => 'AB');
+  /// print(data); // {1: AB, 2: AB, 3: AB}
+  /// data.updateAll((key, value) => key.toString());
+  /// print(data); // {1: 1, 2: 2, 3: 3}
+  /// ```
```

Comment:
I suggest adding a couple of line breaks just to split this up
```
   /// ```dart
   /// final data = {1: 'A', 2: 'B', 3: 'C'};
   /// 
   /// data.updateAll((key, value) => 'AB');
   /// print(data); // {1: AB, 2: AB, 3: AB}
   /// 
   /// data.updateAll((key, value) => key.toString());
   /// print(data); // {1: 1, 2: 2, 3: 3}
   /// ```
```